src/frontend/ts/helpers/chess.ts

```diff
@@ -28,3 +30,26 @@ export function getChessDests (chess: ChessInstance): Map<string, string[]> {
 
   return dests
 }
+
+export function getTally (ship: Ship): String {
+  const { urbit, tallies } = useChessStore()
+
+  let tally: string
+
+  if (ship === `~${urbit.ship}`) {
+    tally = ''
+  } else {
+    if (!tallies.has(ship)) {
+      tally = '0 - 0'
+    } else {
+      const results = tallies.get(ship)
+      const ourWins = results.losses
+      const oppWins = results.wins
```

Comment:
Weren't these `results.losses + (results.draws / 2)` before? Why was that changed to this?

Reply:
Yeah, that `+ Math.floor(results.draws / 2)` was first-draft GPT code carried over from the original commit that I figured was unnecessary while making revisions to this. Unnecessary code was characteristic of its output.

Here's the earlier version:

```typescript
const results = tallies.get(ship)
const ourWins = results.losses + Math.floor(results.draws / 2)
const oppWins = results.wins + Math.floor(results.draws / 2)
const draws = (results.draws % 2) === 1 ? "½" : "";
```

If an opponent's `results` were `{wins: 2, losses: 2, draws: 3}` in the `tallies` map this `getTally` function would halve `3` to get `1.5`, `floor` that to `1`, add that to both `wins` and `losses` then represent the tally as if the results had been `{wins: 3, losses: 3, draws: 3}`, outputting `'3½ - 3½'`. I think the correct tally would be `'2½ - 2½'`

Here I'm logging the input `wins.results`, `losses.results`, `draws.results` and the output `oppWins`, `ourWins`, `draws.results`. (Logged twice as ~dev and ~ser are in the same Firefox window.)

![](https://i.imgur.com/wiMj8zj.png)

Reply:
'3½ - 3½' is the correct tally based on typical chess scoring